src/med_bench/estimation/dml.py

```diff
@@ -0,0 +1,181 @@
+from med_bench.nuisances.propensities import estimate_treatment_probabilities
+from med_bench.nuisances.cross_conditional_outcome import estimate_cross_conditional_mean_outcome_nesting
+
+
+class DoubleMachineLearning(Estimator):
+    """Implementation of double machine learning
+
+    Parameters
+    ----------
+        settings (dict): dictionnary of parameters
+        lbda (float): regularization parameter
+        support_vec_tol (float): tolerance for discarding non-supporting vectors
+            if |alpha_i| < support_vec_tol * lbda then vector is discarded
+        verbose (int): in {0, 1}
+    """
+
+    def __init__(self, procedure : str, density_estimation_method : str, clip : float, trim : float, normalized : bool, sample_split : int, **kwargs):
+        super().__init__(**kwargs)
+
+        self._crossfit = 0
+        self._procedure = procedure
+        self._density_estimation_method = density_estimation_method
+        self._clip = clip
+        self._trim = trim
+        self._normalized = normalized
+        self._sample_split = sample_split
+
+    def resize(self, t, m, x, y):
+        """Resize data for the right shape
+
+        Parameters
+        ----------
+        t       array-like, shape (n_samples)
+                treatment value for each unit, binary
+
+        m       array-like, shape (n_samples)
+                mediator value for each unit, here m is necessary binary and uni-
+                dimensional
+
+        x       array-like, shape (n_samples, n_features_covariates)
+                covariates (potential confounders) values
+
+        y       array-like, shape (n_samples)
+                outcome value for each unit, continuous
+        """
+        if len(y) != len(y.ravel()):
+            raise ValueError("Multidimensional y is not supported")
+        if len(t) != len(t.ravel()):
+            raise ValueError("Multidimensional t is not supported")
+
+        n = len(y)
+        if len(x.shape) == 1:
+            x.reshape(n, 1)
+        if len(m.shape) == 1:
+            m = m.reshape(n, 1)
+
+        if n != len(x) or n != len(m) or n != len(t):
+            raise ValueError(
+                "Inputs don't have the same number of observations")
+
+        y = y.ravel()
+        t = t.ravel()
+
+        return t, m, x, y
+    
+
+    def fit(self, t, m, x, y):
+        """Fits nuisance parameters to data
+
+        """
+        self.fit_score_nuisances(t, m, x, y)
+        t, m, x, y = self.resize(t, m, x, y)
+
+        self.fit_treatment_propensity_x_nuisance(t, x)
+        self.fit_treatment_propensity_xm_nuisance(t, m, x)
+        self.fit_cross_conditional_mean_outcome_nuisance(t, m, x, y)
+        self._fitted = True
+
+        if self.verbose:
+            print("Nuisance models fitted")
+
+
+    def estimate(self, t, m, x, y):
+        """Estimates causal effect on data
+
+        """
+        t, m, x, y = self.resize(t, m, x, y)
+
+        p_x, p_xm = estimate_treatment_probabilities(t,
```

Comment:
Just wondering why `estimate_treatment_probabilities` is not a method of `Estimator` ?